### ISSUE TITLE
Feature request: Add and save keyboard/mouse macros to ARM SoC

### ISSUE THREAD

**Issue by Strykar** [dslul/logiconf#2]
The G402 can save profiles on board the mouse itself, this feature would enable use of the remaining buttons.

**Comment by dslul** [comment]
I'm not sure how to organize the UI (for buttons assignment and macros), any advice?

**Comment by Strykar** [comment]
@dslul Perhaps we can use the same layout as Logitech's Windows software? I haven't used it in a while, but I could pull up screenshots if you like.

EDIT: It would be easy to understand with a photo of the mouse with the button names overlaid on top.
Then, below this we could have just the button names with a text box to record/type mouse/key stroke/combos.

**Comment by dslul** [comment]
I don't think we can reuse the same images in LGS, but there are some in the libratbag project https://github.com/libratbag/libratbag/tree/master/data/gnome
EDIT: I don't like this design that much, so I am more inclined to use images from the old [default theme](https://github.com/libratbag/libratbag/tree/master/data/default), but as they deprecated it, it's missing many devices.

**Comment by Strykar** [comment]
What's wrong with the libratbag SVG?
![logitech-g402](https://user-images.githubusercontent.com/2946372/53682985-91bfec00-3d21-11e9-9696-3d3ec2fa2deb.png)

**Comment by dslul** [comment]
> What's wrong with the libratbag SVG?

Nothing actually, I'm not sure if that design matches the current style of the app, since they were made mostly for gnome programs. I can give it a try though.

**Comment by FFY00** [comment]
Just an heads up, all the libratbag's SVGs were moved to piper.

**Comment by Strykar** [comment]
> 
> Nothing actually, I'm not sure if that design matches the current style of the app, since they were made mostly for gnome programs. I can give it a try though.

Yes please, could we? As https://github.com/libratbag/libratbag/issues/664 has bugs which prevent >1 key macros being set

**Comment by Strykar** [comment]
Is this really stuck because of an appropriate  image? Couldn't the SVG be used as a placeholder and replaced with an image you fancy later?

**Comment by dslul** [comment]
> Is this really stuck because of an appropriate image?

Of course not. The images were never the problem, just a lack of time (and motivation).

**Comment by Strykar** [comment]
@dslul Hey, any chance there's an update on this?

**Comment by dslul** [comment]
> @dslul Hey, any chance there's an update on this?

Not yet, but I hope soon...I could use some help though
